src/charm.py

```diff
@@ -223,13 +271,134 @@ def _on_get_rule_action(self, event: ActionEvent) -> None:
         event.log(f"Successfully fetched rule: {rule_id}")
         event.set_results(rule)
 
-    def _configure_auth_proxy_provider(self, event: HookEvent) -> None:
-        """Create access rules and update config."""
-        pass
+    def _on_auth_proxy_config_changed(self, event: AuthProxyConfigChangedEvent) -> None:
+        if not self._oathkeeper_service_is_running:
+            self.unit.status = WaitingStatus("Waiting for Oathkeeper service")
+            event.defer()
+            return
+
+        if not self._peers:
+            self.unit.status = WaitingStatus("Waiting for peer relation")
+            event.defer()
+            return
+
+        access_rules_locations = list()
+        for rule_type in ("allow", "deny"):
+            rules = self._render_access_rules(
+                rule_type=rule_type,
+                protected_urls=event.protected_urls,
+                allowed_endpoints=event.allowed_endpoints,
+                relation_app_name=event.relation_app_name,
+            )
+
+            if rules:
+                # Push the rules to the container
+                filename = f"{self._oathkeeper_access_rules_path}/access-rules-{event.relation_app_name}-{rule_type}.json"
+                self._container.push(filename, str(rules), make_dirs=True)
+                access_rules_locations.append(filename)
```

Comment:
this should edit the k8s configmap (mounted as a volume) instead of the file

Reply:
As agreed during the daily, it will be addressed in a separate PR